apis/core/v1beta1/flagd_types.go

```diff
@@ -0,0 +1,120 @@
+/*
+Copyright 2022.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta1
+
+import (
+	v1 "k8s.io/api/core/v1"
+	networkingv1 "k8s.io/api/networking/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// FlagdSpec defines the desired state of Flagd
+type FlagdSpec struct {
```

Comment:
I know naming is hard ;) but should we emphasise this a side-car in this naming - `FlagdSidecarSpec` ?

Reply:
Besides, for my understanding, is this created per namespace where we expect special flagd deployment manipulation with ingress and service ? :)

Reply:
I would keep it as `FlagdSpec` in this case, as this contains all the configuration options for a standalone flagd deployment, service and ingress - `FlagdSidecarSpec` would suggest that the options available here are only for a single flagd container in my opinion

Reply:
and yes, the Flagd CRD is namespaced, so the deployment/service/ingress will be created in the same namespace as the `Flagd` resource :)

Reply:
Another option is `FlagdCustomizationSpec` ;)  feel free to ignore.